tests/lib/fakestore/store/store.go

```diff
@@ -684,8 +718,17 @@ func (s *Store) snapActionEndpoint(w http.ResponseWriter, req *http.Request) {
 			return
 		}
 
-		if fn, ok := snaps[name]; ok {
-			essInfo, err := snapEssentialInfo(w, fn, snapID, bs)
+		var snapPath string
+		var foundSnap bool
+		if a.Channel != "" {
+			snapPath, foundSnap = snaps[fmt.Sprintf("%s|%s", name, a.Channel)]
+		}
+		if !foundSnap {
+			snapPath, foundSnap = snaps[name]
+		}
```

Comment:
Very good question. I suppose I did it like that because other tests are not setting channels correctly, that are matching the model, or the `--channel=...` parameter to refresh or install. And I suppose it is rarely "latest/stable" the tests actually look for. But I do not remember. That is probably a task to look at at some point to clean-up the broken tests.

Reply:
I will add a comment